lib/kramdown/parser/gfm.rb

```diff
@@ -38,8 +38,10 @@ def add_hard_line_breaks(element)
             omit_trailing_br = (Kramdown::Element.category(element) == :block && element.children[-1] == child &&
                                 lines[-1].empty?)
             lines.each_with_index do |line, index|
-              children << Element.new(:text, (index > 0 ? "\n#{line}" : line))
-              children << Element.new(:br) if index < lines.size - 2 ||
+              new_element_options = { :location => element.options[:location] + index }
```

Comment:
There is still a problem with this in that `element.options[:location]` is always 1...

While tested with simple examples as here:

```markdown
one
two
three
```

it works as index is doing it's job, using `'` in text breaks the whole code :cry: